### PR TITLE
HoloRepository Demo Setup

### PR DESCRIPTION
**Pull request by boonwj**
This PR is for the creation of a HoloRepository Demo setup.

It includes the steps, demo data and changes to the tools necessary to set it up. The following provides an overall view of what is included in.

- Demo data for EHR FHIR server
- Demo data for HoloStorage FHIR server
- Demo data for HoloStorage Blob service
- Additional feature for `az_blob_tool` to allow upload of a folder of files
- Additional feature for `fhir_api_tool` to upload files containing individual FHIR resource
- Change `Holostorage Accessor` to retain file extension when storing data in blob storage 
- Change Postman query collection to be related to demo data 

TODO: UI end is not tested with the demo data. @fanbomeng97  or @nbckr, to look into it

### CODE REVIEW

**Comment by boonwj** [comment]
Oh yes, the demo FHIR EHR service is currently spun up as https://holoehr-demo.azurewebsites.net/

File: HoloStorageAccessor/internal/holostorageaccessor/api_default.go [hunk 1]
```diff
@@ -329,7 +332,8 @@ func HologramsPost(c *gin.Context) {
 		return
 	}
 
-	err = UploadHologramToBlobStorage(newHologramAPISpec.Hid, hologramFileIO)
+	hologramFilename := newHologramAPISpec.Hid + ".glb"
```
Comment (nielsboecker) [inline HoloStorageAccessor/internal/holostorageaccessor/api_default.go:337]: Name in `Content-Disposition` header is different

File: Misc/holorepo_demo/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# HoloRepository Demo
+This folder contains demo data to setup HoloRepository and the README will document the steps to spin up the necessary infrastructure for HoloRepository
+
+## Prerequisites
+- An Azure subscription
+
+## Contents
+The following describes the contents in this
```
Comment (nielsboecker) [inline Misc/holorepo_demo/README.md]: this... directory?

File: Misc/tools/fhir_api_tool/fhir_api_tool.py [hunk 3]
```diff
@@ -52,6 +54,21 @@ def upload_bundle(self, fhir_json: str):
                 url = urljoin(self._base_url, entry["request"]["url"])
                 self._request(url, "POST", entry["resource"])
 
+    def _upload_resource(self, content: dict):
+        if "id" not in content or "resourceType" not in content:
+            logging.error(
+                "Singular FHIR resource does not contain Id or resourceType...Skipping"
```
Comment (nielsboecker) [inline Misc/tools/fhir_api_tool/fhir_api_tool.py]: lowercase `id` to avoid confusion

File: Misc/holorepo_demo/README.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# HoloRepository Demo
+This folder contains demo data to setup HoloRepository and the README will document the steps to spin up the necessary infrastructure for HoloRepository
```
Comment (nielsboecker) [inline Misc/holorepo_demo/README.md]: Maybe the name of this sub-dir could be more expressive. Maybe `project_setup`? `setup_infrastructure`? Or be fun (i think you had this idea with the branch name) and go `holo_world`

File: Misc/holorepo_demo/README.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+# HoloRepository Demo
```
Comment (nielsboecker) [inline Misc/holorepo_demo/README.md]: The main `README` should have new section that redirects readers here w.r.t. to setting up everything

**Review by nielsboecker — COMMENTED** [review]
Code and docs look fab! 🏎 
Will verify data from UI perspective soon.

**Review by nielsboecker — COMMENTED** [review]
Confirming that the data is loading in UI:
image

it looks pretty good and patients have diifferent numbers of imaging studies:
image


and also holograms:
image

Looks very good. Minor points:
* hologram titles should be according to what the file contains (if the upload tools are too generic for that, maybe manually hack it, as this is what we are going to show people)
* error when I try to download holograms:
```
{"errorCode":"500","errorMessage":"-\u003e github.com/Azure/azure-storage-blob-go/azblob.newStorageError, C:/Users/niels/go/src/github.com/Azure/azure-storage-blob-go/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=BlobNotFound) =====\nDescription=The specified blob does not exist.\nRequestId:ed2d0608-d01e-00c2-65ce-588b02000000\nTime:2019-08-22T09:49:04.7314607Z, Details: \n   Code: BlobNotFound\n   GET https://holoblob.blob.core.windows.net/holograms/h102.glb?timeout=61\n   Authorization: REDACTED\n   User-Agent: [Azure-Storage/0.7 (go1.12.7; Windows_NT)]\n   X-Ms-Client-Request-Id: [e29f61bf-afd3-49d6-70d8-1610f7ff3f76]\n   X-Ms-Date: [Thu, 22 Aug 2019 09:49:04 GMT]\n   X-Ms-Version: [2018-11-09]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 404 The specified blob does not exist.\n   Content-Length: [215]\n   Content-Type: [application/xml]\n   Date: [Thu, 22 Aug 2019 09:49:04 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Error-Code: [BlobNotFound]\n   X-Ms-Request-Id: [ed2d0608-d01e-00c2-65ce-588b02000000]\n   X-Ms-Version: [2018-11-09]\n\n\n"}
```

**Comment by boonwj** [comment]
Sorry, just noticed the comments here. It seems the error with the accessor and comments on the poor data is due to the accessor being configured to the old blobstorage. 

I will post the configurations required to connect to the accessor in teams.

**Comment by nielsboecker** [comment]
Alright. I cannot get my React dev server to run atm, but I actually had the same 500 when I did the PR for Fanbo, and then I got valid data when I changed the config to the demo endpoint. So should be fine.

**Comment by nielsboecker** [comment]
image
Hologram titles are better, but I would prefer them not having the "Hologram" prefix (minor issue).

The other thing is that I'm still getting the 500 though, when trying the download endpoint. Btw, is it okay to test the delete endpoint or will it mess you up on your end? :)
```
http://51.141.224.131/api/v1/holograms/h101/download

=> 
{
"errorCode": "500",
"errorMessage": "-> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /go/src/github.com/Azure/azure-storage-blob-go/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=BlobNotFound) =====\nDescription=The specified blob does not exist.\nRequestId:d483c229-301e-0016-74e8-5b48b0000000\nTime:2019-08-26T08:31:22.2762751Z, Details: \n   Code: BlobNotFound\n   GET https://holoblobdemo.blob.core.windows.net/holograms/h101?timeout=61\n   Authorization: REDACTED\n   User-Agent: [Azure-Storage/0.7 (go1.12.7; linux)]\n   X-Ms-Client-Request-Id: [f008fdde-0528-461e-45f3-bd5e75604e57]\n   X-Ms-Date: [Mon, 26 Aug 2019 08:31:22 GMT]\n   X-Ms-Version: [2018-11-09]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 404 The specified blob does not exist.\n   Content-Length: [215]\n   Content-Type: [application/xml]\n   Date: [Mon, 26 Aug 2019 08:31:21 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Error-Code: [BlobNotFound]\n   X-Ms-Request-Id: [d483c229-301e-0016-74e8-5b48b0000000]\n   X-Ms-Version: [2018-11-09]\n\n\n"
}
````

File: HoloStorageAccessor/internal/holostorageaccessor/api_default.go (hunk 1 above)
Comment (boonwj) [reply HoloStorageAccessor/internal/holostorageaccessor/api_default.go:337]: ah yes, hologramFilename is the filename stored in the blob storage. The one in the content-disposition is the filename provided when somebody downloads the file. Will leave a comment it to make it clearer.

File: Misc/holorepo_demo/README.md (hunk 2 above)
Comment (boonwj) [reply Misc/holorepo_demo/README.md]: changed to `The following describes the contents provided in the demo.`

File: Misc/tools/fhir_api_tool/fhir_api_tool.py (hunk 3 above)
Comment (boonwj) [reply Misc/tools/fhir_api_tool/fhir_api_tool.py]: changed

File: Misc/holorepo_demo/README.md (hunk 4 above)
Comment (boonwj) [reply Misc/holorepo_demo/README.md]: ok holo_world it is

**Comment by boonwj** [comment]
> image
> Hologram titles are better, but I would prefer them not having the "Hologram" prefix (minor issue).
> 
> The other thing is that I'm still getting the 500 though, when trying the download endpoint. Btw, is it okay to test the delete endpoint or will it mess you up on your end? :)
> 
> ```
> http://51.141.224.131/api/v1/holograms/h101/download
> 
> => 
> {
> "errorCode": "500",
> "errorMessage": "-> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /go/src/github.com/Azure/azure-storage-blob-go/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=BlobNotFound) =====\nDescription=The specified blob does not exist.\nRequestId:d483c229-301e-0016-74e8-5b48b0000000\nTime:2019-08-26T08:31:22.2762751Z, Details: \n   Code: BlobNotFound\n   GET https://holoblobdemo.blob.core.windows.net/holograms/h101?timeout=61\n   Authorization: REDACTED\n   User-Agent: [Azure-Storage/0.7 (go1.12.7; linux)]\n   X-Ms-Client-Request-Id: [f008fdde-0528-461e-45f3-bd5e75604e57]\n   X-Ms-Date: [Mon, 26 Aug 2019 08:31:22 GMT]\n   X-Ms-Version: [2018-11-09]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 404 The specified blob does not exist.\n   Content-Length: [215]\n   Content-Type: [application/xml]\n   Date: [Mon, 26 Aug 2019 08:31:21 GMT]\n   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]\n   X-Ms-Error-Code: [BlobNotFound]\n   X-Ms-Request-Id: [d483c229-301e-0016-74e8-5b48b0000000]\n   X-Ms-Version: [2018-11-09]\n\n\n"
> }
> ```

This is because the accessor that is currently deployed is different from the one in this branch. It does not add the '.glb' extension when looking for holograms in the blobstorage.

**Review by nielsboecker — APPROVED** [review]
Great!

**Pull request merged**